sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BulkProcessingOptions.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.batch.BatchRequestResponseConstants;
+import com.azure.cosmos.util.Beta;
+import reactor.core.publisher.Flux;
+
+import java.time.Duration;
+
+/**
+ * Encapsulates options for executing a bulk. This is immutable once
+ * {@link CosmosAsyncContainer#processBulkOperations(Flux, BulkProcessingOptions)} is called, changing it will have no affect.
+ */
+@Beta(Beta.SinceVersion.V4_9_0)
+public final class BulkProcessingOptions<TContext> {
+    private int maxMicroBatchSize = BatchRequestResponseConstants.MAX_OPERATIONS_IN_DIRECT_MODE_BATCH_REQUEST;
+    private int maxMicroBatchConcurrency = BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_CONCURRENCY;
+    private Duration maxMicroBatchInterval = Duration.ofMillis(BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_INTERVAL_IN_MILLISECONDS);
+    private final TContext batchContext;
+
+    public BulkProcessingOptions(TContext batchContext) {
```

Comment:
do we really need the TContext here? From the content of the rest of the class and its methods, it does not look like we have any particular logic to be considered based on this TContext.

Reply:
we'll need it - that is the way how the caller can pass through state needed to update bookmarks etc. We discussed this when we iterated over it in the public surface area discussions. Please leave it for now - we can add samples and then iterate if really needed.